provider/gce/environ_broker.go

```diff
@@ -231,32 +257,20 @@ func getMetadata(args environs.StartInstanceParams, os jujuos.OSType) (map[strin
 // the new instances and returns it. This will always include a root
 // disk with characteristics determined by the provides args and
 // constraints.
-func getDisks(spec *instances.InstanceSpec, cons constraints.Value, ser, eUUID string, daily bool) ([]google.DiskSpec, error) {
+func getDisks(spec *instances.InstanceSpec, cons constraints.Value, ser, eUUID string, imageURLBase string) ([]google.DiskSpec, error) {
 	size := common.MinRootDiskSizeGiB(ser)
 	if cons.RootDisk != nil && *cons.RootDisk > size {
 		size = common.MiBToGiB(*cons.RootDisk)
 	}
-	var imageURL string
-	os, err := series.GetOSFromSeries(ser)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	switch os {
-	case jujuos.Ubuntu:
-		if daily {
-			imageURL = ubuntuDailyImageBasePath
-		} else {
-			imageURL = ubuntuImageBasePath
-		}
-	case jujuos.Windows:
-		imageURL = windowsImageBasePath
-	default:
-		return nil, errors.Errorf("os %s is not supported on the gce provider", os.String())
+	if imageURLBase == "" {
+		return nil, errors.NotValidf("imageURLBase must be set")
 	}
+	imageURL := imageURLBase + spec.Image.Id
+	logger.Infof("fetching disk image from %v", imageURL)
```

Comment:
Decided to add some logging in case people ever got confused when they were trying different options.